User/bri/53452 middleware 1.3.39 (#76)
fiskaltrust/release-notes: middleware/middleware-1.3.39.md

---
slug: /release-notes/middleware/1.3.39
title: Version 1.3.39
---

# fiskaltrust.Middleware 1.3.39 (Germany)
_September 05, 2022_

> In this version of the Middleware,we made it possible to run the German middleware on an ARM device, we've also fixed a bug concerning Fail-transactions not removing the failed start transactions on the FailedStartTransaction table in the database.

Reviewer: ```suggestion
In this version of the Middleware, we made it possible to run the German middleware on an ARM device, we've also fixed an issue in which explicit Fail-transaction calls in failed mode were not removing the failed start transactions in the database.
```

:::caution

Version 1.3 of the Middleware is meant for the German market only, customers in Austria and France should continue to use version 1.2. We will unify these experiences in an upcoming version.

:::

## Feature: ARM support for the German middleware

> We now offer support for customers using ARM devices and wish to implement our German middleware. These are the configuration working on ARM devices :

Reviewer: ```suggestion
We now offer support for customers using ARM devices and wish to implement our German middleware. These are the configurations working on ARM devices:
```

> - SQLITE

Reviewer: ```suggestion
- SQLite
```

Reviewer 2: IMO not completely correct, sorry 😅 fiskaly and Cryptovision should also work, as well as the other Queues.

Maybe it would be nicer to highlight that _most_ things are working, except for some infrequently used TSEs, and TSEs that don't have support for Arm in their SDK. These don't work IMO:
- Epson SCU
- Diebold Nixdorf SCU
- Swissbit Hardware SCU on Arm64 (32-bit works, there's no SDK für Arm64, unfortuantely)

The remaining packages should be good :)

> - Swissbit hqrdware TSE

Reviewer: ```suggestion
- Swissbit Hardware TSE
```

- SwissbitCloud TSE

> ## Bug Fix Explicit fail-transaction-receipt doesn't remove failed start transactions

Reviewer: ```suggestion
## Bug Fix Explicit fail-transaction-receipt doesn't remove failed start transactions in failed mode
```

> We´ve solved a bug where in some cases the Fail-transaction receipt did not remove the failed start transactions from the FailedStartTransaction table in the database, resulting in an exception thrown. Now when an explicit failed is sent, it closes all failed start transaction in failed mode.

Reviewer: ```suggestion
We´ve solved a bug where in some cases the Fail-transaction receipt did not remove the failed start transactions from the FailedStartTransaction table in the database, when beeing in failed mode. Now when an explicit failed is sent, it closes all failed start transactions.
```

## Affected packages
Packages not listed here were not updated, as we decided to not increase the version of unchanged packages. All packages with versions greater or equal to 1.3.1 are compatible with each other (it is e.g. possible to use _fiskaltrust.Middleware.SCU.Swissbit.1.3.1_ with the new queue packages).

- _fiskaltrust.Middleware.Queue.EF v1.3.36_
- _fiskaltrust.Middleware.Queue.SQLite v1.3.39_
- _fiskaltrust.Middleware.Queue.MySQL v1.3.37_
- _fiskaltrust.Middleware.SCU.DE.DeutscheFiskal v1.3.38_
- _fiskaltrust.Middleware.SCU.DE.SwissbitCloud v1.3.39_
- _fiskaltrust.Middleware.SCU.DE.Swissbit v1.3.39_

> 

Reviewer: are these the correct package versions?

Author: packages have been updated based on the list given

---
slug: /release-notes/middleware/1.3.39
title: Version 1.3.39
---

# fiskaltrust.Middleware 1.3.39 (Germany)
_September 05, 2022_

> In this version of the Middleware,we made it possible to run the German middleware on an ARM device,moreover, we implemented an automatic retry when updating the FCC, we've also fixed a bug concerning Fail-transactions not removing the failed start transactions on the FailedStartTransaction table in the database.

Reviewer: ```suggestion
In this version of the Middleware, we made it possible to run the German middleware on an ARM device, moreover,  we implemented an automatic retry when updating the FCC, we've also fixed a bug concerning Fail-transactions not removing the failed start transactions on the FailedStartTransaction table in the database.
```

:::caution

Version 1.3 of the Middleware is meant for the German market only, customers in Austria and France should continue to use version 1.2. We will unify these experiences in an upcoming version.

:::

## Feature: ARM support for the German middleware

> We now offer support for customers using ARM devices and wish to implement our German middleware. Most packages should work on ARM processors, except for these ones , because they do not offer ARM support in their software development kits:

Reviewer: ```suggestion
We now offer support for customers using ARM devices and wish to implement our German middleware. Most packages should work on ARM processors, except for these ones, because they do not offer ARM support in their software development kits:
```
